src/main/java/org/myrobotlab/framework/repo/ServiceData.java

```diff
@@ -286,62 +435,59 @@ public boolean save(String filename) {
 
       return true;
     } catch (Exception e) {
-      Logging.logError(e);
+      log.error("service data saving threw for {}", filename, e);
     }
 
     return false;
   }
 
-  // TWO LEVELS !!! 1. Run-time checking & Build-time checking
-  // Built-time checking
-  // build time has access to the repo - can cross check dependencies to make
-  // sure they are in the library
-  //
-  // Runtime checking
-  // for all Peers - do ALL THERE TYPES CURRENTLY EXIST ???
-  // FIXME - TODO - FIND
-
-  public List<Category> getCategories() {
-    ArrayList<Category> categories = new ArrayList<Category>();
-    for (Category category : categoryTypes.values()) {
-      categories.add(category);
+  /**
+   * Start at root and build all the meta data - add 
+   * 
+   * @param serviceName
+   * @param serviceType
+   * @return
+   */
+  public static Plan getPlan(String serviceName, String serviceType) {
+    
+    Plan root = new Plan();

```

Comment:
This is really just a set (treeset)  .. it's a little difficult to justify a whole new class just to get you a toString method..

Reply:
Its deleted in a future pr.  Was "left" here to reduce number of changes